crates/trigger-http/src/spin.rs

```diff
@@ -27,11 +27,27 @@ impl HttpExecutor for SpinHttpExecutor {
             component_id
         );
 
-        let (instance, store) = engine.prepare_instance(component_id).await?;
+        let (instance, mut store) = engine.prepare_instance(component_id).await?;
         let EitherInstance::Component(instance) = instance else {
             unreachable!()
         };
 
+        if let Some(authority) = req.uri().authority() {
```

Comment:
This has grown to a big enough block that it rather breaks up the flow of the function - consider pulling it out into a `set_http_origin_from_request(&mut store, &req)` (or otherwise meaningfully named) function.